src/machine/machine_rp2040_watchdog.go

```diff
@@ -4,24 +4,67 @@ package machine
 
 import (
 	"device/rp"
-	"runtime/volatile"
-	"unsafe"
 )
 
-type watchdogType struct {
-	ctrl    volatile.Register32
-	load    volatile.Register32
-	reason  volatile.Register32
-	scratch [8]volatile.Register32
-	tick    volatile.Register32
+// Watchdog provides access to the hardware watchdog available
+// in the RP2040.
+var Watchdog = &watchdogImpl{}
+
+const (
+	// WatchdogMaxTimeout in milliseconds (approx 8.3s).
+	//
+	// Nominal 1us per watchdog tick, 24-bit counter,
+	// but due to errata two ticks consumed per 1us.
+	// See: Errata RP2040-E1
+	WatchdogMaxTimeout = (rp.WATCHDOG_LOAD_LOAD_Msk / 1000) / 2
+)
+
+type watchdogImpl struct {
+	// The value to reset the counter to on each Update
+	loadValue uint32
 }
 
-var watchdog = (*watchdogType)(unsafe.Pointer(rp.WATCHDOG))
+// Configure the watchdog.
+//
+// This method should not be called after the watchdog is started and on
+// some platforms attempting to reconfigure after starting the watchdog
+// is explicitly forbidden / will not work.
```

Comment:
This seems to have been copy-pasted for each individual watchdog implementation, should we have a single watchdog.go implementation which calls `Watchdog.configure(cfg)`?, same goes for `Start`

Reply:
I don't think we've done that anywhere else, just for docs.  Would be kind of a big change to remove public methods from platform-specific files.  Happy to do it if we think it's beneficial - the public function should easily get inlined.

Reply:
I think we should leave it as it is now, for the reasons @kenbell just mentioned.